File: TPs/TP02/Soma/Soma.c
#include <stdio.h>
#include <stdlib.h>

int Soma(int num){
    
    if(num/10 == 0){//compara se o resultado é 0

        return num;

    }else{

        int somar = num%10; //guarda os restos das divisões

        return somar + Soma(num/10); //recursão para soma
    }
}

int main(){

    int num;
    char digito[20]; //armazena um numero ou uma string

    scanf("%s", digito);

    while(!(digito[0] == 'F' && digito[1] == 'I' && digito[2] == 'M' && digito[3] == '\0')){

        num = atoi(digito); //atoi converte string para int

        int resul = Soma(num);

        printf("%d\n", resul);

        scanf("%s", digito);
    }


    return 0;
}

File: TPs/TP03/Soma.java
import java.util.Scanner;
public class Soma {
    public static int Somar(int num){
        if(num/10 == 0){ //compara se o resultado é 0
            return num;
        }else{
            int n = num/10; //guarda o resultado das divisoes
            return num%10 + Somar(n);  //recursão para soma
        }
    }
    public static void main(String[]args){
        Scanner scanner = new Scanner(System.in);
        String digito = scanner.nextLine();
        int num;
        

        while(digito.charAt(0) != 'F' && digito.charAt(1) != 'I' && digito.charAt(2) != 'M'){
            num = Integer.parseInt(digito); // transforma String em inteiro
            int resul = Somar(num);
            System.out.println(resul);
            digito = scanner.nextLine();
        }

        scanner.close();
    }
}